scripts/tuple_tree_generator/tuple_tree_generator/definition/defaults.py

```diff
@@ -0,0 +1,5 @@
+# This file is distributed under the MIT License. See LICENSE.md for details.
+
+DEFAULT_STRUCT_NAMESPACE = "model::generated"
+DEFAULT_ENUM_NAMESPACE = "model"
+DEFAULT_USER_NAMESPACE = "model"
```

Comment:
Can we have a test for a non-model tuple-tree-like data structure?
We have plans to take certain things out of the model, and having a simple example on how to create such a thing would help a lot.

Reply:
I refactored the cmake invoking `tuple_tree_generator`, added a test that does a round-trip (object1 -> yaml -> object2, assert object1 == object2) and can serve as an example for most things `tuple_tree_generator` can do.
Also, now `tuple_tree_generator` takes the namespace to use as a parameter.